### ISSUE TITLE
How to integrate Spooldir CVS connector with Schema registry?

### ISSUE THREAD

**Issue by ashutakGG** [jcustenborder/kafka-connect-spooldir#115]
Hello,

The documentation says "If you couple this with the Avro converter and Schema Registry by Confluent, you will be able to process CSV, Json, or TSV files to strongly typed Avro data in real time."

I was not able to find any example of doing that.

I guess, that I must some kind of converter or transformer here. Could you please advice which one or should it be custom one?

**Comment by jcustenborder** [comment]
Hi  @ashutakGG 

The cool part about the connect framework is I don't have to think about serialization. This connector takes the CSV data and converts it to Kafka Connect structs that are strongly typed. From there it hands the data to the connect framework which uses a Converter to serialize the data. The [Avro Converter](https://docs.confluent.io/current/schema-registry/connect.html) will handle serializing the data. All you need to do is configure the Avro Converter for your workers or for the individual connector. I recommend you do this at the worker level. Reopen this  issue if you have any questions.

**Comment by aeldaly-tc** [comment]
We are building an internal service to allow any of our internal services to place documents into Elastic Search. We are heavily leveraging schema-registry for formatting the fields to their correct type in ES. 

To process creating documents in real-time, we have a main topic, configured to hook into schema-registry and uses the RecordNameStrategy. All systems produce to that one topic, and as long as they've registered the schema beforehand, all documents end up in ES with the correct types out of the box, and with minimal configuration.

We want to support the ability for any of those systems to upload a csv file via ftp and have the contents end up in ES. How do we achieve the same outcome with csv files? We're having issues figuring out how to configure this bit. So let's start with the first one:

Any system can drop a csv file of unknown fields into the folder. So how do we parse it? Our first solution was to create an endpoint that takes the subject name of the schema that they want to use, go to schema registry and use the schema to generate the connector and supplying the `value.schema` config option. Problem is, `value.schema` wants a kafka connect schema, not an avro schema format. So that didn't work....

Basically, we feel like from the documentation and the comment above that this feature is built in and integration with schema registry is there, but we cannot really wrap our heads around how to do it.

Thanks for your assistance :)

**Comment by jcustenborder** [comment]
@aeldaly-tc The spooldir csv connector is designed to search a directory for files that match a certain pattern. In order to pull off what you're trying to do you'd need to run more than one connector. That connector would search for a particular pattern. You could still use the RecordNameStrategy but each connector will only support a single schema at a time.

**Comment by sachin-rajwade** [comment]
my connector config is

CREATE SOURCE CONNECTOR `the-spool-connector` WITH (
   "connector.class"='com.github.jcustenborder.kafka.connect.spooldir.SpoolDirCsvSourceConnector',
   "topic"='DES-product_inventory',
   "input.path"='/data/unprocessed/ncs/product_inventory',
   "finished.path"='/data/processed/ncs/product_inventory',
   "error.path"='/data/errored/ncs/product_inventory',
   "input.file.pattern"='.*\\.csv',
   "key.converter"='org.apache.kafka.connect.storage.StringConverter',
   "value.converter"='io.confluent.connect.avro.AvroConverter',
   "value.converter.schema.registry.url"= 'http://192.168.15.91:8089/',
   "csv.first.row.as.header"='true',
   "schema.generation.enabled"='true',
   "enhanced.avro.schema.support"=true,
   "transforms"='castTypes',
   "transforms.castTypes.type"='org.apache.kafka.connect.transforms.Cast$Value',
   "transforms.castTypes.spec"='uuId:string,tenantId:string,eventName:string,productId:string,inventory:int64'      
);

But it fails at the kafka level with avro fail

**Comment by jcustenborder** [comment]
That should work fine. The full call stack would help figure out why it failed.

**Comment by sachin-rajwade** [comment]
> That should work fine. The full call stack would help figure out why it failed.

Can i use my schema while sending the csv to kafka ? eg XXX-value in schema registry
